src/python_inspector/utils_pypi.py

```diff
@@ -1595,11 +1595,32 @@ def fetch_links(
             url, _, _sha256 = anchor_tag["href"].partition("#sha256=")
             if "data-requires-python" in anchor_tag.attrs:
                 python_requires = anchor_tag.attrs["data-requires-python"]
+            url = resolve_relative_url(package_url, url)  # Resolve relative URL
             links.append(Link(url=url, python_requires=python_requires))
         # TODO: keep sha256
         return links
 
 
+def resolve_relative_url(package_url, url):
+    """
+    Resolve a relative URL using the package URL.
+
+    Args:
+        package_url (str): The base URL of the package.
+        url (str): The URL to be resolved.
+
+    Returns:
+        str: The resolved URL.
+    Examples:
+        >>> resolve_relative_url("https://example.com/package", "../path/file.txt")
+        'https://example.com/path/file.txt'
+    """
+    if not url.startswith(("http://", "https://")):
+        base_url = "/".join(package_url.split("/")[:-1])  # Extract base URL
```

Comment:
Nitpicking: have you considered using instead https://docs.python.org/3/library/urllib.parse.html?highlight=urlparse#urllib.parse.urlparse  for a more robust and explicit handling of URLs? splitting and slicing is OK, but I feel that the standard library may be better here and will make end of line comments no longer needed (and I would prefer regualr comments, not end-of-line)